src/rust/engine/async_semaphore/src/lib.rs

```diff
@@ -77,7 +77,7 @@ impl AsyncSemaphore {
     res
   }
 
-  async fn acquire(&self) -> Permit<'_> {
+  pub async fn acquire(&self) -> Permit<'_> {
```

Comment:
This was private from way back in the day when Future combinators were widely used. In an async-await world, it's easy to see how to use it correctly.